### PR TITLE
Add persisting scroll position to left navigation after page refresh

### PR DESCRIPTION
**Pull request by DonOmalVindula**
## Purpose
Makes the scrollbar in the left navigation remember its last scroll position and then scroll back to that position whenever page refresh occurs (such as switching to another page) and thereby improving the UX. 
This also improves the UX by offsetting the scroll bar by few pixels when coming from to accommodate the space taken by the header. This will allow the users to see target the heading they clicked on without being obstructed by the top header.

## Goals

1. Preserve the scroll position of the left navigation when the page is reloaded so that the user will not have to scroll back to the same position if he/she visits a different link or refreshes the page.

2. Offsetting the scroll position by few pixels to prevent the target headers being obstructed by the header when a user is coming from a target link.

Resolves https://github.com/wso2/product-is/issues/11908
Resolves https://github.com/wso2/product-is/issues/11909

## Approach

1.  This was achieved using a Javascript code added to `en/docs/assets/js/theme.js`

2.  The current scroll position of the left navigation was extracted and stored in `session storage` just before the reload occurs.

3.  After the reload, the stored variable was taken from session storage and reassigned as the new scroll position of the left navigation, persisting the scroll position after reload.

4. Scroll offset was handled by adding an offset value of  `50` to the scroll when a user is coming from another page via target links.

## Test environment

- mkdocs - 1.1.2
- mkdocs-material - 4.1.1
- mkdocs-material-extensions - 1.0.1
- pymdown-extensions - 5.0

### CODE REVIEW

File: en/docs/assets/js/theme.js [hunk 1]
```diff
@@ -305,3 +305,28 @@ window.addEventListener('scroll', function() {
...
+
+// Preserving scroll position of the left nav on reload or switching pages
+let timer = null;
+var leftSidebarScrollPos = 0;
+var leftSidebar = document.querySelector(".md-sidebar--primary > .md-sidebar__scrollwrap");
+leftSidebar.addEventListener('scroll', (function (e) {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+        leftSidebarScrollPos = leftSidebar.scrollTop;
+        console.log(leftSidebarScrollPos);
```
Comment (dewniMW) [inline en/docs/assets/js/theme.js:317]: why is this log required?

File: en/docs/assets/js/theme.js [hunk 2]
```diff
@@ -305,3 +305,28 @@ window.addEventListener('scroll', function() {
...
+    },100)
+}));
+
+// saving the scroll position in session storage
+window.onbeforeunload = function () {
+    sessionStorage.clear("navScrollPos");
+    sessionStorage.setItem("navScrollPos", leftSidebarScrollPos);
+}
+// retriving the scroll position on reload
```
Comment (dewniMW) [inline en/docs/assets/js/theme.js:326]:
```suggestion
// Retrieving the scroll position on reload.
```

File: en/docs/assets/js/theme.js [hunk 3]
```diff
@@ -305,3 +305,28 @@ window.addEventListener('scroll', function() {
...
+var leftSidebar = document.querySelector(".md-sidebar--primary > .md-sidebar__scrollwrap");
+leftSidebar.addEventListener('scroll', (function (e) {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+        leftSidebarScrollPos = leftSidebar.scrollTop;
+        console.log(leftSidebarScrollPos);
+    },100)
+}));
+
+// saving the scroll position in session storage
```
Comment (dewniMW) [inline en/docs/assets/js/theme.js:321]:
```suggestion
// Saving the scroll position in session storage.
```

File: en/docs/assets/js/theme.js [hunk 4]
```diff
@@ -305,3 +305,28 @@ window.addEventListener('scroll', function() {
 window.addEventListener("hashchange", function () {
     window.scrollTo(window.scrollX, window.scrollY - 40, 'smooth');
 });
+
+// Preserving scroll position of the left nav on reload or switching pages
```
Comment (dewniMW) [inline en/docs/assets/js/theme.js:309]:
```suggestion
// Preserving scroll position of the left nav on reload or switching pages.
```

File: en/docs/assets/js/theme.js [hunk 5]
```diff
@@ -305,3 +305,27 @@ window.addEventListener('scroll', function() {
...
+        leftSidebarScrollPos = leftSidebar.scrollTop;
+    },100)
+}));
+
+// Saving the scroll position in session storage.
+window.onbeforeunload = function () {
+    sessionStorage.clear("navScrollPos");
+    sessionStorage.setItem("navScrollPos", leftSidebarScrollPos);
+}
+// Retriving the scroll position on reload.
```
Comment (dewniMW) [inline en/docs/assets/js/theme.js:325]:
```suggestion
// Retrieving the scroll position on reload.
```

File: en/docs/assets/js/theme.js [hunk 6]
```diff
@@ -305,3 +305,27 @@ window.addEventListener('scroll', function() {
...
+    sessionStorage.clear("navScrollPos");
+    sessionStorage.setItem("navScrollPos", leftSidebarScrollPos);
+}
+// Retrieving the scroll position on reload.
+window.onload = function () {
+    var prevScrollPos = parseInt(sessionStorage.getItem("navScrollPos"));
+    if(prevScrollPos) {
+        leftSidebar.scrollTop = prevScrollPos;
+    }
+}
```
Comment (dewniMW) [inline en/docs/assets/js/theme.js:331]: add a newline to the end of a file

**Review by NipuniBhagya — APPROVED** [review]

**Pull request merged**